modules/common/src/main/scala-2.10/notebook/util/Deps.scala

```diff
@@ -5,13 +5,19 @@ import sbt._
 import scala.util.Try
 
 object Deps extends java.io.Serializable {
+  private val PATTERN_MODULEID_1 = """^\s*([^% ]+)\s*%\s*([^% ]+)\s*%\s*([^% ]+)\s*$""".r
+  private val PATTERN_MODULEID_2 = """^\s*([^% ]+)\s*%\s*([^% ]+)\s*%\s*([^% ]+)\s*%\s*([^% ]+)\s*$""".r
+  private val PATTERN_COORDINATE_1 = """^\s*([^:/ ]+)[:/]([^: ]+):([^: ]+)\s*$""".r
+
   def parseInclude(s: String): Option[ModuleID] = {
     s.headOption.filter(_ != '-').map(_ => s.dropWhile(_ == '+').trim).flatMap { line =>
-      line.replaceAll("\"", "").split("%").toList match {
-        case List(g, a, v) =>
-          Some(g.trim % a.trim % v.trim % "compile")
-        case List(g, a, v, p) =>
-          Some(g.trim % a.trim % v.trim % p.trim)
+      line.replaceAll("\"", "") match {
+        case PATTERN_MODULEID_1(g, a, v) =>
+          Some(g % a % v % "compile")
+        case PATTERN_MODULEID_2(g, a, v, p) =>
+          Some(g % a % v % p)
+        case PATTERN_COORDINATE_1(g, a, v) =>
```

Comment:
I just wonder if we couldn't just stick with the sbt like pattern? Since spark-packages is just a simple maven hosted on bintray, it'll work because maven repo are already supported.